charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -111,54 +111,27 @@ Return if ingress is stable.
   {{- eq (include "ingress.apiVersion" .) "networking.k8s.io/v1" -}}
 {{- end -}}
 
-
-{{- define "opentelemetry-collector.daemonset-podAnnotations" -}}
-{{- if eq .Values.mode "daemonset" }}
-  {{- if .Values.podAnnotations }}
-  {{- .Values.podAnnotations | toYaml }}
-  {{- end }}
-{{- else if .Values.agentCollector.podAnnotations }}
-{{- .Values.agentCollector.podAnnotations | toYaml }}
-{{- end }}
-{{- end }}
-
-{{- define "opentelemetry-collector.deployment-podAnnotations" -}}
-{{- if eq .Values.mode "deployment" }}
-  {{- if .Values.podAnnotations }}
-  {{- .Values.podAnnotations | toYaml }}
-  {{- end }}
-{{- else if .Values.standaloneCollector.podAnnotations }}
+{{- define "opentelemetry-collector.podAnnotations" -}}
+{{- if and .Values.podAnnotations .Values.mode  }}
+{{- .Values.podAnnotations | toYaml }}
+{{- else if and .Values.standaloneCollector.enabled .Values.standaloneCollector.podAnnotations }}
```

Comment:
This change breaks backwards compatibility with .Values.agentCollector.podAnnotations